v2/src/features/UploadList/components/FileCard.tsx

```diff
@@ -0,0 +1,117 @@
+import React from 'react';
+import Card from '@mui/material/Card';
+import CardActions from '@mui/material/CardActions';
+import CardContent from '@mui/material/CardContent';
+import Tooltip from '@mui/material/Tooltip';
+import IconButton from '@mui/material/IconButton';
+import OpenInNewIcon from '@mui/icons-material/OpenInNew';
+import ContentCopyIcon from '@mui/icons-material/ContentCopy';
+import EmailOutlinedIcon from '@mui/icons-material/EmailOutlined';
+import CloseIcon from '@mui/icons-material/Close';
+import ReplayIcon from '@mui/icons-material/Replay';
+import LinearProgress from '@mui/material/LinearProgress';
+import CardTextField from './CardTextField';
+import { styled } from '@mui/material/styles';
+
+import Clipboard from 'libs/clipboard';
+import { SugarFileState } from '../types';
+import settings from 'settings';
+
+type Props = {
+  data: SugarFileState;
+  onCancel: () => void;
+  onRetry: () => void;
+};
+
+const StyledCard = styled(Card)(({ theme }) => ({
+  width: '90%',
+  marginBottom: theme.spacing(1),
+}));
+
+export default function FileCard({ data, onCancel, onRetry }: Props) {
+  const { file, shareableLink, upload, error } = data;
+  const isUploading = upload.state && upload.progress < 100;
+  const isError = error.state !== null;
+
+  // Show notification in helperText of a TextField
+  const [notification, setNotification] = React.useState<string | null>(null);
+
+  const notify = (text: string, millis = 2000) => {
+    setNotification(text);
+    setTimeout(() => setNotification(null), millis);
+  };
+
+  const copyToClipboard = () => {
+    Clipboard.copy(
+      shareableLink,
+      () => notify('Copied to clipboard!'),
+      () => notify('Copy failed, ensure clipboard permissions.')
+    );
+  };
+
+  return (
+    <StyledCard>
+      <CardContent sx={{ padding: 1, margin: 0 }}>
+        <CardTextField
+          {...data}
+          onClick={copyToClipboard}
+          notification={notification}
+        />
+
+        {isUploading && (
+          <LinearProgress
+            variant={upload.progress === 100 ? 'determinate' : 'indeterminate'}
+            value={upload.progress}
+            color={error.state ? 'error' : 'primary'}
+          />
+        )}
+      </CardContent>
+      <CardActions disableSpacing sx={{ p: 0, mx: 1 }}>
+        {error.state === 'retriable' && (
+          <Tooltip title='Retry'>
+            <IconButton aria-label='retry' size='small' onClick={onRetry}>
+              <ReplayIcon />
+            </IconButton>
+          </Tooltip>
+        )}
+        <Tooltip title='Cancel'>
```

Comment:
It would be great to move this guy to the far right and leave the action buttons alone on the left.

Reply:
Issue created to address this separatedly